src/Lower.cpp

```diff
@@ -220,6 +221,13 @@ Stmt lower(const vector<Function> &outputs, const string &pipeline_name, const T
     s = simplify(s);
     debug(2) << "Lowering after partitioning loops:\n" << s << "\n\n";
 
+    if (t.features_any_of({Target::HVX_64, Target::HVX_128})) {
```

Comment:
Actually Dillon has convinced me that the right place for this is indeed inside CodeGen_Hexagon